libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -101,6 +103,16 @@ std::string IRGenerator::generate(
 	std::map<ContractDefinition const*, std::string_view const> const& _otherYulSources
 )
 {
+	auto notTransient = [](VariableDeclaration const* _varDeclaration) {
+		solAssert(_varDeclaration);
+		return _varDeclaration->referenceLocation() != VariableDeclaration::Location::Transient;
+	};
+
+	solUnimplementedAssert(
+		ranges::all_of(_contract.stateVariables(), notTransient),
+		"Transient storage variables are not supported."
```

Comment:
```suggestion
		"Transient storage variables are not supported in via IR codegen."
```

Also, add another semanticTests with `compileViaYul: true` that triggers this assert (just to make sure we're not allowing for IR codegen). I'll get rid of it in my PR once the IR codegen pipeline is supported.

Reply:
I think it's going to be more trouble than it's worth, specially since you will remove the test already.
The problem here is that semantic tests are not prepared or intended to handle codegen errors. Ultimately they are going to end up at:

https://github.com/ethereum/solidity/blob/5dbaa13bf7699db57f964e786ccff838da1bd80d/test/libsolidity/SolidityExecutionFramework.cpp#L71-L84

Then we can't move forward because, in the last line in this snippet, when we call `m_compiler.object(contractName);` we hit this assertion:

https://github.com/ethereum/solidity/blob/5dbaa13bf7699db57f964e786ccff838da1bd80d/libsolidity/interface/CompilerStack.cpp#L941

I think it's best to just manually check that it generates the `UnimplementedFeatureError`, which I  have done now.

Reply:
My bad - I thought we were handling unimplemented errors gracefully in semantic tests. In that case, you can add a CLI test (a copy of the one you already have which produces proper assembly) - `--evm-version cancun --no-cbor-metadata --via-ir --optimize --asm --debug-info none` or better yet `--evm-version cancun --no-cbor-metadata --ir-optimized --debug-info none`.

This test case will obviously result in a failure, and then I can just update the expectation on the via-IR side.

Reply:
Done.